client/src/entrypoints/admin/bulk-actions.js

```diff
@@ -139,14 +143,18 @@ function addBulkActionListeners() {
       checkedState.numPages++;
       el.addEventListener('change', SelectBulkActionsCheckboxes);
     });
-  document.querySelector(`.${BULK_ACTION_SELECT_ALL_CHECKBOX_TH}`).addEventListener('change', SelectBulkActionsFilter);
+  document.querySelectorAll(`.${BULK_ACTION_SELECT_ALL_CHECKBOX_TH} input`).forEach(el => {
+    el.addEventListener('change', SelectBulkActionsFilter);
+  });
   document.querySelectorAll(`.${BULK_ACTION_FILTERS_CLASS}`).forEach(
     elem => elem.addEventListener('click', FilterEventListener)
   );
-  document.querySelectorAll(`.${BULK_ACTION_CHOICES_DIV} > ul > li > a`).forEach(
+  document.querySelectorAll(`.${BULK_ACTION_CHOICES_DIV} ul > li > a`).forEach(
```

Comment:
I don't really love this dependence on the html structure to this extent. We could consider using a data attribute on the choice itself, or at least simplify this selector.